tests/scripts/thread-cert/Cert_5_5_07_SplitMergeThreeWay.py

```diff
@@ -111,6 +117,80 @@ def test(self):
             if addr[0:4] != 'fe80':
                 self.assertTrue(self.nodes[ROUTER1].ping(addr))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER_1']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_3 = pv.vars['ROUTER_3']
+        leader_pkts = pkts.filter_wpan_src64(LEADER)
+        router1_pkts = pkts.filter_wpan_src64(ROUTER_1)
+        router2_pkts = pkts.filter_wpan_src64(ROUTER_2)
+        router3_pkts = pkts.filter_wpan_src64(ROUTER_3)
+
+        # Step 2: The Leader and Router_1 MUST send properly formatted MLE Advertisements
+        leader_pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+        _lpkts = leader_pkts.copy()
+        _lpkts.filter_mle_cmd(MLE_ADVERTISEMENT).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, ROUTE64_TLV} == set(p.mle.tlv.type))
+
+        router1_pkts.range(leader_pkts.index).filter_mle_cmd(MLE_ADVERTISEMENT).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, ROUTE64_TLV} == set(p.mle.tlv.type))
+
+        # Step 4: Each router forms a partition with the lowest possible partition ID
+        # Step 5: Router_1 MUST send MLE Parent Requests and MUST make two separate attempts
+        router1_pkts.filter_mle_cmd(MLE_PARENT_REQUEST).must_next().must_verify(
+            lambda p: {MODE_TLV, CHALLENGE_TLV, SCAN_MASK_TLV, VERSION_TLV} == set(
+                p.mle.tlv.type) and p.mle.tlv.scan_mask.r == 1 and p.mle.tlv.scan_mask.e == 1)
+        lreset_start = router1_pkts.index
+        router1_pkts.filter_mle_cmd(MLE_PARENT_REQUEST).must_next().must_verify(
+            lambda p: {MODE_TLV, CHALLENGE_TLV, SCAN_MASK_TLV, VERSION_TLV} == set(
+                p.mle.tlv.type) and p.mle.tlv.scan_mask.r == 1 and p.mle.tlv.scan_mask.e == 1)
+
+        # Step 7: Router_1 MUST attempt to attach to any other Partition
+        # within range by sending a MLE Parent Request.
+        router1_pkts.filter_mle_cmd(MLE_PARENT_REQUEST).must_next().must_verify(
+            lambda p: {MODE_TLV, CHALLENGE_TLV, SCAN_MASK_TLV, VERSION_TLV} == set(p.mle.tlv.type))
+        lreset_stop = router1_pkts.index
+
+        # Step 3: The Leader MUST stop sending MLE advertisements.
+        p = leader_pkts.range(lreset_start, lreset_stop)
+        p.filter_mle_cmd(MLE_ADVERTISEMENT).must_not_next()
```

Comment:
We usually use `p` for a packet. We never use `p` for a `PacketFilter`.

Seems `p` is not necessary.
```suggestion
        leader_pkts.range(lreset_start, lreset_stop).filter_mle_cmd(MLE_ADVERTISEMENT).must_not_next()
```